waku/v2/protocol/waku_store/client.nim

```diff
@@ -40,28 +43,45 @@ proc new*(T: type WakuStoreClient,
           store: MessageStore): T = 
   WakuStoreClient(peerManager: peerManager, rng: rng, store: store)
 
+proc sendHistoryQueryRPC(w: WakuStoreClient, req: HistoryQuery, peer: RemotePeerInfo): Future[HistoryResult] {.async, gcsafe.} =
 
-proc query*(w: WakuStoreClient, req: HistoryQuery, peer: RemotePeerInfo): Future[WakuStoreResult[HistoryResponse]] {.async, gcsafe.} =
   let connOpt = await w.peerManager.dialPeer(peer, WakuStoreCodec)
   if connOpt.isNone():
     waku_store_errors.inc(labelValues = [dialFailure])
-    return err(dialFailure)
+    return err(HistoryError(kind: HistoryErrorKind.PEER_DIAL_FAILURE, address: $peer))
+  
   let connection = connOpt.get()
 
-  let rpc = HistoryRPC(requestId: generateRequestId(w.rng), query: req)
-  await connection.writeLP(rpc.encode().buffer)
 
-  var message = await connection.readLp(MaxRpcSize.int)
-  let response = HistoryRPC.decode(message)
+  let reqRpc = HistoryRPC(requestId: generateRequestId(w.rng), query: req.toRPC())
+  await connection.writeLP(reqRpc.encode().buffer)
 
-  if response.isErr():
-    error "failed to decode response"
+
+  let buf = await connection.readLp(MaxRpcSize.int)
+  let respDecodeRes = HistoryRPC.decode(buf)
+  if respDecodeRes.isErr():
     waku_store_errors.inc(labelValues = [decodeRpcFailure])
-    return err(decodeRpcFailure)
+    return err(HistoryError(kind: HistoryErrorKind.BAD_RESPONSE, cause: decodeRpcFailure))
+
+  let respRpc = respDecodeRes.get()
+
+
+  # Disabled ,for now, since the default response is a possible case (no messages, pagesize = 0, error = NONE(0))
```

Comment:
Can we create a tracking issue for this?

Reply:
It already exists: #1159

Reply:
Okay